text/0140-record-feature-flags-on-error.md

```diff
@@ -0,0 +1,45 @@
+- Start Date: 2024-09-04
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/140
+- RFC Status: draft
+
+# Summary
+
+As feature flags are evaluated in a customer's application we will collect those flags and their results, store them in-memory, and push the values to Sentry when an error occurs. The flags can be used to further understand what went wrong during the session.
+
+# Motivation
+
+To enable users to debug application errors with a complete picture of their application's state.
+
+# Options Considered
+
+We will collect flag evaluations and hold them in-memory.  On error event the flags will be placed into the `contexts` object on the `event` and sent to Sentry.
+
+## Transport
+
+The flags will be represented by this data structure during transport. The `flag` key is the name of the flag and the `result` key is the evaluation result returned by the customer's application.
+
+```json
+{
+    "contexts": {
+        "flags": [
+            {"flag": "abc", "result": true},
+            {"flag": "def", "result": false}
+        ]
+    }
+}
+```
+
+## Public Interface
```

Comment:
Might need a `reset` or `checkpoint` method for back-end applications.  For example an HTTP server, some flags are called at startup and others are called per request.  It would be wrong to see feature flags from a previous request in the current request's scope.

If we're storing on the `scope` maybe this is handled already?

Reply:
We are moving the SDKs to a new scope model, JavaScript is already there and Python is next. Meaning, we don't have hubs anymore, but instead a global scope and an isolation scope. The isolation scope is "isolated" per request.

Reply:
Perfect.

Reply:
fwiw that would work with the Hub model too. Since the original flags resolved during startup would be available to the cloned hubs (and its scope) used to serve requests. 

For example, I had to clear breadcrumbs on aspnet because the bootstrapping logs (sometimes days or months old) would show up on all web requests events and got confusing to some folks